src/coreclr/jit/lowerxarch.cpp

```diff
@@ -6335,38 +6335,49 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
                 {
                     if ((intrinsicId >= NI_FMA_MultiplyAdd) && (intrinsicId <= NI_FMA_MultiplySubtractNegatedScalar))
                     {
-                        bool supportsRegOptional = false;
+                        bool     supportsOp1RegOptional = false;
+                        bool     supportsOp2RegOptional = false;
+                        bool     supportsOp3RegOptional = false;
+                        unsigned resultOpNum            = 0;
+                        LIR::Use use;
 
-                        if (IsContainableHWIntrinsicOp(node, op3, &supportsRegOptional))
+                        if (BlockRange().TryGetUse(node, &use))
                         {
-                            // 213 form: op1 = (op2 * op1) + [op3]
-                            MakeSrcContained(node, op3);
+                            resultOpNum = node->GetResultOpNumForFMA(use.User(), op1, op2, op3);
```

Comment:
It's interesting that we only call this if we get a `use`. If there is no use, I'd think we still would want to do the `IsLastUse` checks, so perhaps this should be something like:
```
LIR::Use use;
GenTree* user = nullptr;

if (BlockRange().TryGetUse(node, &use))
{
    user = use.User();
}

resultOpNum = node->GetResultOpNumForFMA(user, op1, op2, op3);
```

?

Reply:
And then if there is no use, `GetResultOpNumForFMA()` could return the default `0`.